app/uk/gov/hmrc/helptosave/actors/EligibilityStatsHandler.scala

```diff
@@ -107,42 +78,41 @@ class EligibilityStatsHandlerImpl extends EligibilityStatsHandler with Logging {
          || Reason | Channel  |  Count   |
          |+--------+----------+----------+\n""".stripMargin
 
-    var report = table.foldLeft[String](header) {
-      case (row: String, element: (String, List[(String, Int)])) ⇒
-        val a = element._2.foldLeft[String](row) {
-          case (x: String, y: (String, Int)) ⇒
-            x.concat(f.format(element._1, y._1, y._2))
-        }
+    def getStringFrom(reason: String) = {
+      if (reason === "6") {
+        "UC"
+      } else if (reason === "7") {
+        "WTC"
+      } else if (reason === "8") {
+        "UC&WTC"
+      } else {
+        reason
+      }
+    }
 
-          def getTotalByReason =
-            table
-              .filter(p ⇒ p._1 === element._1)
-              .flatMap(_._2)
-              .map(_._2)
-              .sum
+    var report = table.foldLeft[String](header) {
+      case (acc, curr) =>
+        val r = curr._2.foldLeft[String]("") {
```

Comment:
its a list of list , not sure if we can create rows by using only one loop here (currently i used two `foldLeft`s ) ?